substrate/frame/bounties/src/lib.rs

```diff
@@ -1009,6 +1046,61 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 
 		Ok(())
 	}
+
+	/// Helper function to calculate the bounty storage deposit.
+	fn calculate_bounty_deposit(
+		description: &BoundedVec<u8, T::MaximumReasonLength>,
+	) -> BalanceOf<T, I> {
+		T::BountyDepositBase::get() +
+			T::DataDepositPerByte::get() * (description.len() as u32).into()
+	}
+
+	/// Helper function to poke the deposit reserved for proposing a bounty.
+	///
+	/// Returns true if the deposit was updated and false otherwise.
+	fn poke_bounty_deposit(
+		bounty_id: BountyIndex,
+		who: T::AccountId,
+	) -> Result<bool, DispatchError> {
+		let mut bounty = Bounties::<T, I>::get(bounty_id).ok_or(Error::<T, I>::InvalidIndex)?;
+		let bounty_description =
+			BountyDescriptions::<T, I>::get(bounty_id).ok_or(Error::<T, I>::InvalidIndex)?;
+		// ensure that the bounty status is proposed.
+		ensure!(bounty.status == BountyStatus::Proposed, Error::<T, I>::UnexpectedStatus);
+
+		// ensure that bounty proposer is the origin for this call.
+		ensure!(bounty.proposer == who, Error::<T, I>::NotProposer);
+
+		let new_bond = Self::calculate_bounty_deposit(&bounty_description);
+		let old_bond = bounty.bond;
+		if new_bond == old_bond {
```

Comment:
Might be nicer to write this as `match new_bond.cmp { Ordering::Less => {}, Ordering::Greater => {}, Ordering::Equal => {} } `.